### ISSUE TITLE
Support Vector Machine with Pulp

### ISSUE THREAD

**Issue by Benji90210** [coin-or/pulp#140]
Hi,
I'm trying to compare SVM solve with Pulp vs Scikit-learn using the following LP problem:
![equation](https://user-images.githubusercontent.com/30179790/28216933-49697dc8-68ab-11e7-8365-ae55e1118a51.jpg)
Unfortunately, Pulp return some value for **b** and **Eta**, but **Alpha** and **a** are stuck to zero.
I'm question the way I implemented it as I can't find the constrains with the Kernel matrix in the LP file...  

# SVM implementation in Pulp vs Scikit-learn

    import numpy as np
    import matplotlib.pyplot as plt
    from sklearn.svm import SVR
    from sklearn.metrics.pairwise import rbf_kernel
    from pulp import *
    
    def CompPulpSlk():

    N = 100
    x = np.linspace(0, 2 * np.pi, N)
    y = np.sin(x) + np.random.random(N) * 1
    x.resize(N, 1)
    K = rbf_kernel(x, gamma=5)

    # define de problem for Pulp

    # define the hyper parameters
    C = 10
    eps = 0.1

    # define the variables
    RANGE = list(range(N))
    alpha = pulp.LpVariable.dicts("alpha", (RANGE,), cat='Continuous')
    b = pulp.LpVariable("b", cat='Continuous')
    a = pulp.LpVariable.dicts("a", (RANGE,), lowBound=0, cat='Continuous')
    Eta = pulp.LpVariable.dicts("Eta", (RANGE,), lowBound=0, cat='Continuous')

    prob = LpProblem("Constrained SVM", LpMinimize)

    # implementing the constrains to Pulp
    for i in range(N):
        prob += lpDot(alpha, K[i]) + b - y[i] >= -Eta[i]
        prob += lpDot(alpha, K[i]) + b - y[i] <= Eta[i]
        prob += eps >= 0
        prob += eps <= Eta[i]
        prob += alpha[i] <= a[i]
        prob += alpha[i] >= -a[i]

    # call the objective function
    prob += lpSum(a) + C * lpSum(Eta)

    prob.writeLP("MySVM.lp")
    prob.solve()

    print("Status:", LpStatus[prob.status])

    #storing Optimisation result
    varDic = {}
    for v in prob.variables():
        if v.name[:5] == 'alpha' or v.name == 'b':
            varDic[v.name] = v.varValue

    #passing solution (alpha, beta) to proper ndarray
    b = varDic['b']
    del varDic['b']

    alp = np.zeros(N)
    for key in varDic:
        alp[int(key[6:])] = varDic[key]

    #clean up PULP variables
    alpha.clear()
    Eta.clear()
    a.clear()
    varDic.clear()

    #Projection of the solution from Pulp and Sci-learn
    y_pulp = np.zeros(N)
    for i in range(N):
        y_pulp[i] = np.dot(alp, K[i, :]) + b

    svr_rbf = SVR(kernel='rbf', C=1e3, gamma=0.1)
    y_rbf = svr_rbf.fit(x, y).predict(x)

    #display result
    lw = 2
    plt.scatter(x, y, color='darkorange', label='data')
    plt.hold('on')
    plt.plot(x, y_rbf, color='navy', lw=lw, label='SciLearn')
    plt.plot(x, y_pulp, color='cornflowerblue', lw=lw, label='Pulp')
    plt.xlabel('data')
    plt.ylabel('target')
    plt.title('Support Vector Regression')
    plt.legend()
    plt.show()
